### ISSUE TITLE
Parser should complain about unbalanced or nested PRIVATE

### ISSUE THREAD

**Issue by bjourne** [slavapestov/factor#1355]
Like if you write this:

```factor
IN: test22

<PRIVATE

: foo ( a -- b ) ;

PRIVATE>

PRIVATE>
```
There is one to many `PRIVATE>`. Or in:

```factor
IN: test22

<PRIVATE
<PRIVATE
<PRIVATE

: foo ( a -- b ) ;

PRIVATE>
PRIVATE>
PRIVATE>
```
The parser should disallow the nesting.

**Comment by mrjbq7** [comment]
This is the code that begins and ends private blocks from ``vocabs.parser``:

```factor
: begin-private ( -- )
    current-vocab name>> ".private" ?tail
    [ drop ] [ ".private" append set-current-vocab ] if ;

: end-private ( -- )
    current-vocab name>> ".private" ?tail
    [ set-current-vocab ] [ drop ] if ;
```

Instead of ``drop``, you can throw an error.  What would be a useful name for that error?

**Comment by bjourne** [comment]
`unbalanced-private-declaration` perhaps?

**Comment by mrjbq7** [comment]
Fixed in 8744d664673be031acc0ccc393bb2ef62f0caf05.

Already found some b2adcaaac916dd156ba3393dff9c4e968b6aba79 and 24953512a7c3515b92b93329a77792d512688ab9.
